### ISSUE TITLE
Regression in 1.4.0: encoding is not preserved for strings longer than 255 bytes

### ISSUE THREAD

**Issue by casperisfine** [msgpack/msgpack-ruby#200]
I'm digging in the problem right now, and hopefully I should find a fix, but just to let you know @tagomoris 

```ruby
require "msgpack"

string = "a" * 255
p string.bytesize
p string.encoding
p MessagePack.load(MessagePack.dump(string)).encoding
p MessagePack.load(MessagePack.dump(string))


string = "a" * 256
p string.bytesize
p string.encoding
p MessagePack.load(MessagePack.dump(string)).encoding
p MessagePack.load(MessagePack.dump(string))
```

```
255
#<Encoding:UTF-8>
#<Encoding:UTF-8>
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
256
#<Encoding:UTF-8>
#<Encoding:ASCII-8BIT>
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```

This could be related to my recent encoding cleanup.

**Comment by tagomoris** [comment]
Released 1.4.1.
@casperisfine Thank you for the quick report and fix!

**Comment by casperisfine** [comment]
Welcome. My bad for not testing master against our app before you cut the release. I could have saved us all some time :/
